drivers/rtc/rtc_rv8263.c

```diff
@@ -585,9 +603,7 @@ static int rv8263c8_alarm_set_callback(const struct device *dev, uint16_t id,
 	if ((callback == NULL) && (user_data == NULL)) {
 		LOG_DBG("Disable alarm function");
 
-		err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
-					     RV8263C8_BM_ALARM_INT_ENABLE,
-					     RV8263C8_BM_ALARM_INT_DISABLE);
+		err = rv8263c8_disable_alarm(dev);
```

Comment:
`rtc_alarm_set_callback` should control only the alarm callback functionality. When disabling the alarm callback, the alarm should remain enabled until disabling it using `rtc_alarm_set_time()`.

Reply:
However, the alarm should also not be enabled when setting the callback. The documentation is talking about the alarm callback and not the alarm itself.

Reply:
There is definitely lack of documentation and inconsistent implementation regarding the interrupt outputs (I'll open a separate issue on this).

Regardless, calling `rv8263c8_disable_alarm()` here will disable both the alarm and alarm callback, but the documentation of `rtc_alarm_set_callback()` states, that the alarm should remain enabled and only be disabled by calling `rtc_alarm_set_time()`:

> Setting the alarm callback for an alarm, will enable the alarm callback. When the callback for an alarm is enabled, the alarm triggered event will invoke the callback, after which the alarm pending status will be cleared automatically. The alarm will remain enabled until manually disabled using [rtc_alarm_set_time()](https://docs.zephyrproject.org/apidoc/latest/group__rtc__interface.html#ga1fc9f0ec9642dfbf742ea541a20bdad5).
>
> To disable the alarm callback for an alarm, the callback and user_data parameters must be set to NULL. When the alarm callback for an alarm is disabled, the alarm triggered event will set the alarm status to "pending". To check if the alarm status is "pending", use [rtc_alarm_is_pending()](https://docs.zephyrproject.org/apidoc/latest/group__rtc__interface.html#ga531c379c437a91df44a45e95063700e8).